mm2src/coins/eth.rs

```diff
@@ -2775,23 +2791,32 @@ impl EthCoin {
         tx: &[u8],
         swap_contract_address: Address,
         search_from_block: u64,
-    ) -> Result<Option<FoundSwapTxSpend>, String> {
-        let unverified: UnverifiedTransaction = try_s!(rlp::decode(tx));
-        let tx = try_s!(SignedEthTx::new(unverified));
+    ) -> Result<Option<FoundSwapTxSpend>, MmError<FoundSwapTxSpendErr>> {
+        let unverified: UnverifiedTransaction =
+            rlp::decode(tx).map_to_mm(|err| FoundSwapTxSpendErr::Internal(err.to_string()))?;
```

Comment:
Consider implementing `From<Error> for FoundSwapTxSpendErr` if **only** you are sure that such error can be converted the same way in different contexts.
For example,
https://github.com/KomodoPlatform/atomicDEX-API/blob/7cf104abf6005b3b9183a3059e757872315cded8/mm2src/coins/qrc20.rs#L425-L430

Reply:
Not fixed yet